### PR TITLE
python3Packages.ibis-framework: 1.3.0 -> 2.1.1

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Bump `python3Packages.ibis-framework` to 2.1.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
All three commits should be prefixed with the same `python3Packages.ibis-framework:` as they're all related to that package. "chore" probably doesn't mean anything to anyone other than you.

**Comment by cpcloud** [comment]
> All three commits should be prefixed with the same `python3Packages.ibis-framework:` as they're all related to that package. "chore" probably doesn't mean anything to anyone other than you.

Fair enough, cleaning them up now.

**Comment by cpcloud** [comment]
@risicle Is this good to merge?

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 1]
```diff
@@ -1,65 +1,146 @@
...
+
+    runHook preCheck
+
+    pytest --numprocesses auto \
+      ibis/tests \
+      ibis/backends/tests \
+      ibis/backends/{${lib.concatStringsSep "," backends}}/tests \
+      -k '${lib.concatMapStringsSep " and " (test: "not ${test}") disabledTests}'
+
+    runHook postCheck
```
Comment (risicle) [inline pkgs/development/python-modules/ibis-framework/default.nix:135]: Why can `pytestCheckHook` not be used here?

Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix:135]: I suppose it can, I sometimes forget about `pytestFlagsArray` 😅

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15, nixos x86_64 & aarch64.

**Pull request merged**